backend/src/north/north-rest/north-rest.ts

```diff
@@ -61,26 +61,20 @@ export default class NorthREST extends NorthConnector<NorthRESTSettings> {
     const fileStream = createReadStream(filePath);
     form.append('file', fileStream, { filename: base });
 
-    const headers: Record<string, string> = form.getHeaders();
+    const headers = form.getHeaders();
+    const proxyOptions = this.getProxyOptions();
+    const authOptions = this.getAuthorizationOptions();
 
-    // Get proxy agent if needed
-    const proxyAgent = await this.getProxyAgent();
-
-    // Add authorization header if available
-    const authHeader = await this.getAuthorizationHeader();
-    if (authHeader) {
-      headers['Authorization'] = authHeader;
-    }
-
-    let response;
+    let response: ReqResponse;
     try {
-      response = await request(endpoint, {
+      response = await HTTPRequest(endpoint, {
         method: 'POST',
         headers,
         query: queryParams,
         body: form,
-        dispatcher: proxyAgent,
-        signal: AbortSignal.timeout(this.connector.settings.timeout * 1000)
+        auth: authOptions,
+        proxy: proxyOptions,
+        timeout: this.connector.settings.timeout * 1000
       });
     } catch (error) {
       const message = this.getMessageFromError(error);
```

Comment:
Is filestream properly closed, even if we throw an OIBus Error in the catch part?